src/plugin.spec.ts

```diff
@@ -3330,7 +3330,7 @@ describe('MaestroPlugin', () => {
                     m.expectOnce(mc_getAA(data, "Schedules.0.Productions.0"))
                     print mc_getAA(data, "Schedules.0.Productions.0")
                     formatJson(mc_getAA(json, "user"))
-                    print (mc_getString(json, "user.name", "default name"))
+                    print (mc_getString(json?.user?.name, invalid, "default name"))
```

Comment:
I think this will cause runtime exceptions if the item isn't an AA. Like, if json.user is a roDateTime, calling .name on it will throw an exception because you can't use a roDateTime like an AA. 

Is that acceptable?

Reply:
Indeed this is going to throw an error. Not sure how should we proceed  in this case. @georgejecook  what are your thoughts on this ?

Reply:
@TwitchBronBron @georgejecook in the case above could we just put mc_getPath in try catch block and if it throws an error to return the default value ?

Reply:
>. @TwitchBronBron @georgejecook in the case above could we just put mc_getPath in try catch block and if it throws an error just return the default value ?

that sounds reasonable to me.

Reply:
in this case should I make the necessary changes in maestro ?

Reply:
Does this syntax support things like:
```brighterscript
mc_getString(json?.user?.name?.getValue(), invalid, "default name")
```
Because if so, then try/catch doesn't seem like the correct approach...as you'd want those function call exceptions to bubble up.

Reply:
>in this case should I make the necessary changes in maestro ?
yes.

@TwitchBronBron , this function is generally tranpsiled to from asString, in our apps - in that case, the plugin will give a diagnostic error indicating that  method invocations on asXXX() calls are not supported at this time (for the very reason you lay out here)

Reply:
@rdaci23 this is looking good; once we verify this in maestro, with the local compiled plugin, we will be good to merge this.

Reply:
@georgejecook , closing this pr as per our last discussion related to optional chaining operator